lib/connectors_sdk/office365/custom_client.rb

```diff
@@ -255,15 +259,30 @@ def get_root_item(drive_id, fields = [])
         request_endpoint(:endpoint => "drives/#{drive_id}/root", :query_params => query_params)
       end
 
-      def item_children(drive_id, item_id, fields: [], &block)
-        endpoint = "drives/#{drive_id}/items/#{item_id}/children"
-        query_params = transform_fields_to_request_query_params(fields)
-        response = request_endpoint(:endpoint => endpoint, :query_params => query_params)
+      def item_children(drive_id, item_id, fields: [], break_after_page: false, &block)
+        next_link = cursors.delete('item_children_next_link') if break_after_page
+
+        response = if next_link.present?
+                     request_json(:url => next_link)
+                   else
+                     endpoint = "drives/#{drive_id}/items/#{item_id}/children"
+                     query_params = transform_fields_to_request_query_params(fields)
+                     request_endpoint(:endpoint => endpoint, :query_params => query_params)
+                   end
 
+        yielded = 0
         loop do
           response.value.each(&block)
           next_link = response['@odata.nextLink']
```

Comment:
I think you could always just assign `cursors['item_children_next_link'] = response['@odata.netLink']` here, and above always use `next_link = cursors['item_children_next_link']` (line 263) without having to couch these inside of `if break_after_page` checks. We should always use the cursor if there is a cursor, and if there isn't a cursor it means we don't need to use it. Right?

Reply:
I think that'd cause external connector cursors to start showing up in internal connector cursors. With the if-wrapping, it also makes these easy to remove once we no longer need the per-data-source sync pagination logic like this.

Reply:
> I think that'd cause external connector cursors to start showing up in internal connector cursors

Isn't that a good thing? That would make our internal connector syncs better at resuming failed/crashed jobs, right? I definitely think we want a future where internal and external connectors still use the same cursoring (totally fine if that's not feasible today, just looking at this change I thought maybe it was).

Reply:
Ah, these changes were not intended to affect internal connector behavior. While there's some potential value there, I've not thought at all about the effect of including this behavior in internal connector sync logic. Certainly not worth the risk at this particular point, pending the 8.2.0 release.